### DISCUSSION TITLE
Setting US State information automatically

### DISCUSSION THREAD

**Discussion by m0urs** [magicbug/Cloudlog#2558] [General]
I see that many QSOs  in my log with US stations do have the US state field correctly populated. However I see others who do not have that information set. So I am unsure which mechanism is populating the state field? Could someone give me a hint, please?

**Comment by magicbug** [comment]
If you have linked Cloudlog to QRZs Callbook or LoTW it will be coming from either of those

**Comment by m0urs** [comment]
I have linked both. However at least QRZ does not update the US state for me?

I even tested with a manual created QSO. All other information is updated correctly from QRZ, but not the state information?

And if you upload QSOs via the QRZ menu entry it is just uploading but not reading information from QRZ,com and updating the local entry.

Did I miss something?

**Comment by m0urs** [comment]
Here an example:

![image](https://github.com/magicbug/Cloudlog/assets/2588362/236f8947-4ebe-49ad-ba73-2d8c496ff184)

The state information is present in QRZ.com-

**Comment by magicbug** [comment]
When I say QRZ i mean when your typing callsign into /QSO

This data is likely coming from LoTW

**Reply by m0urs** [reply]
Ok, so that means QSOs which are not confirmed sometime later via LOTW do not have the state information updated if the QSOs are either imported or sent via an external program to Cloudlog.

Did I get that right?

If yes, I would like to make an Enhancement Request for some kind of batch updating missing state information from QRZ.com .... :-)

**Reply by magicbug** [reply]
QRZ batch already exists `/index.php/update/check_missing_grid`

**Reply by m0urs** [reply]
Just tested that again. It does update the Grid if it is missing, but it does not update the US state for me!?

**Reply by magicbug** [reply]
Cant remember about state i think it does I'll check but can add it :)

**Comment by AndreasK79** [comment]
@m0urs did you try this button in the advanced logbook?
![image](https://github.com/magicbug/Cloudlog/assets/6977712/1a859cc2-4d37-44d8-95b7-179629c0438d)

**Reply by m0urs** [reply]
Yes, but I am unsure what it should update. It does nothing for my tested entries. At least it does not update the US state information.

**Reply by AndreasK79** [reply]
I see. Guess I need to dig into the code again. I am pretty sure it did work when I implemented it :)

**Reply by AndreasK79** [reply]
It is also written so that it does not overwrite existing info. It should just add the missing stuff.

**Reply by m0urs** [reply]
Thanks. If we could have that also implemented to use it automated in batch (like updating Clublog, QRZ etc.) and not only manually it would be even better :-)
